tools/provision/debian.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+# g++ stuff in wheezy has a bug which has been fixed in gcc 4.7.3 which conveniently is exactly 
+# the version that is not shipped in wheezy
+# See https://gcc.gnu.org/viewcvs/gcc/branches/gcc-4_7-branch/libstdc%2B%2B-v3/include/std/condition_variable?view=patch&r1=189276&r2=193528&pathrev=193528
+
+function patch_condition_variable()
+{
+	TARGET_DIR=/usr/include/c++/4.7
+	pushd $TARGET_DIR
+	
+	if [[ ! -f condition_variable.orig ]]; then		
+		sudo cp condition_variable condition_variable.orig
+		sudo patch -p5 < $FILES_DIR/debian/condition_variable.patch
+
+		log "patched $TARGET_DIR/condition_variable which prevents rocksdb from building"
+	else
+		log "$TARGET_DIR/condition_variable already patched: found a .orig"
+	fi
+	
+	popd
+}
+	
+function main_debian() {
+	sudo apt-get update -y
+	sudo apt-get upgrade -y
+
+	package git-core
+	package wget
+	
+	# Add LLVM to the APT sources	
+	
+	if [[ $DISTRO == "wheezy" ]]; then 
+		set_cc clang
+		set_cxx clang++
+	
+		wget -O - http://llvm.org/apt/llvm-snapshot.gpg.key|sudo apt-key add -
```

Comment:
Is it possible to wrap this change then 2nd update in a condition such that it's not happening on every `make deps` run. This will save considerable time for each PR build! :dancer: